packages/itwinui-css/src/button/button.scss

```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
   position: relative;
   border-radius: $iui-border-radius;
   line-height: $iui-line-height;
-  box-shadow: none;
   font-size: $iui-font-size;
   font-weight: $iui-font-weight-normal;
-  text-decoration: none;
   user-select: none;
   cursor: pointer;
   white-space: nowrap;
-  border: 1px solid transparent;
+  border: 1px solid var(--_iui-button-border-color);
+  background-color: var(--_iui-button-background-color);
   color: var(--_iui-button-text-color);
-  @media (prefers-reduced-motion: no-preference) {
-    transition: color $iui-speed-fast ease-out, background-color $iui-speed-fast ease-out,
-      border-color $iui-speed-fast ease-out;
+  transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out;
+  -webkit-tap-highlight-color: transparent;
+
+  @at-root a#{&} {
+    text-decoration: none;
   }
 
-  &:hover {
-    --_iui-button-text-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-1));
+  // #region Sizing
+  @include iui-button-size;
 
-    text-decoration: none;
+  &[data-iui-button-size='small'] {
+    @include iui-button-size(small);
   }
 
-  @include iui-focus;
+  &[data-iui-button-size='large'] {
+    @include iui-button-size(large);
+  }
+  // #endregion Sizing
+
+  &[aria-disabled='true'] {
+    @include iui-focus;
+  }
 
   &[disabled],
-  &:disabled {
-    @include iui-button-disabled;
+  &:disabled,
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+
+    &,
+    &:hover,
+    &:active {
+      --_iui-button-text-color: var(--iui-text-color-placeholder);
+      --_iui-button-background-color: var(--iui-color-background-disabled);
+      --_iui-button-border-color: var(--iui-color-background-disabled);
+    }
+  }
+}
+
+@mixin iui-button-label {
+  white-space: nowrap;
+  color: inherit;
```

Comment:
I've removed the mixin but left the class names even though they don't have any styling associated.

Reply:
Where is the class name useful? Could we replace those usages with `> *`?

Reply:
Within this button PR it looks like `.iui-button-text` is being styled within `header.scss` and `side-navigation.scss`. I don't think `> *` would work in those instances as is currently because there is also an icon nested within the button.

However, #640 uses [`.iui-header-breadcrumb-button-text`](https://github.com/iTwin/iTwinUI/blob/b5143acda0da28517bff191ec7273e0dd1548fb1/packages/itwinui-css/backstop/tests/header.html#L133) instead.  We could do the same to `side-navigation`.

Reply:
@veekeys, any opinion on this?  [You suggested using `.iui-button-text`](https://github.com/iTwin/iTwinUI/pull/684#discussion_r914013136) to align with other components.

Reply:
Will remove `.iui-button-text` classes.